### PR TITLE
More warning fixes

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Comment by lrknox** [comment]
@seanm I've run formatter resulting in changes to 4 test files.  Shall I commit changes?

**Comment by seanm** [comment]
Commit, sure. But might want to give reviewers more time for merge.

File: test/ros3.c [hunk 1]
```diff
@@ -865,7 +865,7 @@ test_vfd_open(void)
         if (T.which_fapl == FAPL_FILE_ACCESS)
             _fapl_id = fapl_file_access;
         else if (T.which_fapl == FAPL_ROS3_ANON)
-            _fapl_id = fapl_id;
+            _(void) fapl_id;
```
Comment (qkoziol) [inline test/ros3.c]: ?

**Review by qkoziol — APPROVED** [review]
Aside from the one typo, this looks good.

File: test/ros3.c (hunk 1 above)
Comment (seanm) [reply test/ros3.c]: whoops. good catch. A find-replace screwup I guess. Clearly I'm not compiling that file. Do you know what build option I'm missing?

Comment (qkoziol) [reply test/ros3.c]: > whoops. good catch. A find-replace screwup I guess. Clearly I'm not compiling that file. Do you know what build option I'm missing?

I believe that the "read-only s3" ("ros3") VFD needs to talk to an S3 server, but it's enabled with the --enable-ros3-vfd configure flag.

File: c++/test/tarray.cpp [hunk 2]
```diff
@@ -182,7 +182,7 @@ test_array_compound_array()
         // Check the array dimensions
         for (ii = 0; ii < ndims; ii++)
             if (rdims1[ii] != tdims1[ii]) {
-                TestErrPrintf("Array dimension information doesn't match!, rdims1[%d]=%zd, tdims1[%d]=z%d\n",
+                TestErrPrintf("Array dimension information doesn't match!, rdims1[%d]=%lld, tdims1[%d]=%lld\n",
```
Comment (gnuoyd) [inline c++/test/tarray.cpp]: I'm not sure about `%lld`.... There's definitely a disagreement in signedness. `"%" PRIuHSIZE` will always work to format an `hsize_t`.

File: c++/test/tarray.cpp [hunk 3]
```diff
@@ -459,7 +459,7 @@ test_array_info()
             for (ii = 0; ii < ndims; ii++)
                 if (rdims1[ii] != tdims1[ii]) {
                     TestErrPrintf(
-                        "Array dimension information doesn't match!, rdims1[%d]=%zd, tdims1[%d]=z%d\n", ii,
+                        "Array dimension information doesn't match!, rdims1[%d]=%llu, tdims1[%d]=z%llu\n", ii,
```
Comment (gnuoyd) [inline c++/test/tarray.cpp:465]: Looks like that z is a typo?

File: hl/src/H5IM.c [hunk 4]
```diff
@@ -169,6 +169,11 @@ find_palette(H5_ATTR_UNUSED hid_t loc_id, const char *name, H5_ATTR_UNUSED const
     if (name == NULL)
         return -1;
 
+    /* Shut compiler up */
+    (void)loc_id;
+    (void)ainfo;
+    (void)op_data;
+
```
Comment (gnuoyd) [inline hl/src/H5IM.c:176]: The H5_ATTR_UNUSED didn't do the trick? Maybe it needs to go between the type and the parameter name? If so, funny if the use above doesn't produce an error.

File: c++/test/tarray.cpp (hunk 2 above)
Comment (seanm) [reply c++/test/tarray.cpp]: Indeed the signedness was and is wrong, will fix.

File: c++/test/tarray.cpp (hunk 3 above)
Comment (seanm) [reply c++/test/tarray.cpp:465]: Good catch. I fixed one, but apparently there's this one too.

**Comment by lrknox** [comment]
I ran the clang formatter to make the format check pass.  There was a conflict for line 868 of test/ros3.c;  please check to see whether it is correct.

File: hl/src/H5IM.c (hunk 4 above)
Comment (seanm) [reply hl/src/H5IM.c:176]: In fact it does. I removed the redundancy.

**Comment by seanm** [comment]
@lrknox oh dear, I just pushed my fixes to what @gnuoyd pointed out, before seeing your comment.  I think it maybe overwrote your formatting changes... let's see if anyone has any other necessary changes, then could you reformat again?

**Comment by lrknox** [comment]
Sure, no problem.

**Review by derobins — APPROVED** [review]

File: test/ros3.c (hunk 1 above)
Comment (seanm) [reply test/ros3.c]: Fixed.

**Pull request merged**